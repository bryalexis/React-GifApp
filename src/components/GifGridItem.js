import React from 'react'

export const GifGridItem = ({id, title, url}) => {
  return (
    <div className='gifItem animate__animated animate__backInLeft'>
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}
