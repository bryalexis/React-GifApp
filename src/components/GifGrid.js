import React from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    // const [images, setImages] = useState([])
    
    // useEffect( () => {
    //     getGifs(category)
    //         .then(setImages);
    // }, [category]) // Si la categor√≠a cambia se actualiza el componente

    const {data:images, loading} = useFetchGifs(category);

    return (
        <>
            <h3 className='animate__animated animate__fadeIn'>{category}</h3>
            {loading && <p className='animate__animated animate__flash'>Loading...</p>}
            <div className='gifItem-grid'> 
            {
                images.map( img => (
                    <GifGridItem key={img.id} {...img} />
                ))
            }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
