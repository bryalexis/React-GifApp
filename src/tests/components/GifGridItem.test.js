import React from 'react'
import { shallow } from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem'

describe('Pruebas en <GifGridItem />', () => {
    const title = 'Title';
    const url = 'https://localhost/uwu.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);   

    test('Debe mostrar el contenido correctamente', () => { 
        expect( wrapper ).toMatchSnapshot(); 
    }) 

    test('Debe tener un parrafo con el title', () => { 
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('Debe tener la imagen con url y alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('AnimaciÃ³n de carga', () => { 
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__backInLeft')).toBe(true);
    })
})