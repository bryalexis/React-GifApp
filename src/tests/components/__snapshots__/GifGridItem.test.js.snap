// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar el contenido correctamente 1`] = `
<div
  className="gifItem animate__animated animate__backInLeft"
>
  <img
    alt="Title"
    src="https://localhost/uwu.jpg"
  />
  <p>
    Title
  </p>
</div>
`;
